### PR TITLE
Fix for the shm_per_client_regex shmem object

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

This is a bugfix candidate for https://github.com/pi-hole/FTL/issues/868 Whether it fixes this bug or not, it is still good to have this.

### CODE REVIEW

**Comment by DL6ER** [comment]
Rebased on latest `development`

File: src/shmem.c [hunk 1]
```diff
@@ -193,24 +193,29 @@ static pthread_mutex_t create_mutex(void) {
...
 	dns_cache = (DNSCacheData*)shm_dns_cache.ptr;
 
-	realloc_shm(&shm_strings, counters->strings_MAX, false);
+	realloc_shm(&shm_strings, counters->strings_MAX, sizeof(char), false);
 	// strings are not exposed by a global pointer
 
+	// Total number of regex filters
+	const unsigned int num_regex_tot = counters->num_regex[REGEX_BLACKLIST] +
+	                                   counters->num_regex[REGEX_WHITELIST];
+	realloc_shm(&shm_per_client_regex, counters->clients, num_regex_tot * sizeof(bool), false);
```
Comment (DL6ER) [inline src/shmem.c:217]: Adding this line is the actual bugfix, the rest is merely making debug messages more useful in the future.

**Comment by DL6ER** [comment]
Rebased on latest `development`

**Comment by dschaper** [comment]
Were there any discourse topics associated with this PR?

**Comment by DL6ER** [comment]
I don't think so.

**Review by dschaper — APPROVED** [review]

**Pull request merged**